documentation/quick-start.md

```diff
@@ -0,0 +1,236 @@
+
+The simplest `GET` request:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+const response = await got(url);
+```
+
+The call returns a <code>Promise<[Response](3-streams.md#response-1)></code>. If the body contains JSON, it can be retrieved directly:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+const data = await got(url).json();
+```
+
+The similar <code>[promise.text](1-promise.md#promisetext)</code> method returns plain text.
+
+All `got` methods accept an options object for passing extra configuration, such as headers:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	headers: {
+		'Custom-Header': 'Quick start',
+	},
+	timeout: {
+		send: 3500
+	},
+};
+
+const data = await got(url, options).json();
+```
+
+A `POST` request is very similar:
+
+```js
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	json: {
+		documentName: 'Quick Start',
+	},
+};
+
+const data = await got.post(url, options);
+```
+
+The request body is passed in the options object. The `json` property will automatically set headers accordingly. Custom headers can be added exactly as above.
+
+## Using streams
+
+The [Stream API](3-streams.md) allows to leverage [Node.js Streams](https://nodejs.dev/learn/nodejs-streams) capabilities:
+
+```js
+import {open} from 'node:fs/promises';
+import {pipeline} from 'node:stream/promises';
+import got from 'got';
+
+const url = 'https://httpbin.org/anything';
+
+const options = {
+	json: {
+		documentName: 'Quick Start',
+	},
+};
+
+const gotStream = got.stream.post(url, options);
+
+const fileDescriptor = await open('anything.json', 'w');
+const outStream = await fileDescriptor.createWriteStream();
+
+try {
+	await pipeline(gotStream, outStream);
+} catch (error) {
+	console.error(error);
+}
+```
+
+## Options
+
+Options can be set at the client level and reused in subsequent queries:
+
+```js
+import got from 'got';
+
+const options = {
+	prefixUrl: 'https://httpbin.org',
+	headers: {
+		Authorization: getTokenFromVault(),
+	},
+};
+
+const client = got.extend(options);
+
+export default client;
+```
+
+Some noticeable common options are:
+- [`searchParams`](2-options.md#searchparams): A query string object.
+- [`prefixUrl`](2-options.md#prefixurl): Prepended to query paths. Paths must be relative to prefix, i.e. not begin with a `/`.
+- [`method`](2-options.md#method): The HTTP method name.
+- [`headers`](2-options.md#headers): Query headers.
+- [`json`](2-options.md#json): JSON body.
+- [`form`](2-options.md#form): A form query string object.
+
+See documentation for other [options](2-options.md#options).
+
+## Errors
+
+Both Promise and Stream APIs throw errors with metadata.
+
+```js
+import got from 'got';
+
+try {
+	const data = await got.get('https://httpbin.org/status/404');
+} catch (error) {
+	console.error(error.message);
```

Comment:
A better example would be: `error.response.statusCode`.

Reply:
done